bfg-library/src/main/scala/com/madgag/git/bfg/model/Commit.scala

```diff
@@ -59,7 +60,8 @@ case class CommitArcs(parents: Seq[ObjectId], tree: ObjectId) {
 }
 
 object CommitNode {
-  def apply(c: RevCommit): CommitNode = CommitNode(c.getAuthorIdent, c.getCommitterIdent, c.getFullMessage, c.getEncoding)
+  def apply(c: RevCommit): CommitNode = CommitNode(c.getAuthorIdent, c.getCommitterIdent, c.getFullMessage,
+      try c.getEncoding catch {case e: IllegalCharsetNameException => Constants.CHARSET})
```

Comment:
Possibly a refinement would be to split on commas and take the first encoding that we can decode - which might be different to the `UTF-8` of `Constants.CHARSET` - but so far as I can tell, nobody has ever expected the `encoding` header in Git commit messages to contain commas. The implementation in JGit certainly doesn't expect it:

https://github.com/eclipse/jgit/blob/ccc89977/org.eclipse.jgit/src/org/eclipse/jgit/util/RawParseUtils.java#L764-L800

...and C-Git ultimately uses `iconv`, which doesn't seem to expect comma separated values either: http://www.gnu.org/savannah-checkouts/gnu/libiconv/documentation/libiconv-1.13/iconv_open.3.html